src/rulesets/Base/base.jl

```diff
@@ -26,6 +26,12 @@ function rrule(::typeof(real), z::Number)
     return (real(z), real_pullback)
 end
 
+# Conversions to Float
+
+@scalar_rule float(x::Real) true
```

Comment:
This is also true for complex i think
```suggestion
@scalar_rule float(x::Union{Real,Complex}) true
```

Reply:
Is there a reason it needs a type constraint? e.g. if `float` can be called on the primal, then isn't it always safe for us to say the derivative is one? If it can't be called, the frule/rrule will error anyways.

Reply:
true